server/app/services/applicant/ApplicantService.java

```diff
@@ -167,6 +167,12 @@ public ReadOnlyApplicantProgramService getReadOnlyApplicantProgramService(
         application.getApplicantData(), programDefinition, baseUrl);
   }
 
+  /** Get a {@link ReadOnlyApplicantProgramService} from applicant data and a program definition. */
+  public ReadOnlyApplicantProgramService getReadOnlyApplicantProgramService(
```

Comment:
**suggestion (non-blocking):** since this isn't used outside of this class consider making it `private`

Reply:
Good call, updated!